pkg/networkservice/chains/usecase/usecase_test.go

```diff
@@ -46,294 +47,264 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/kernel"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/counter"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/inject/injecterror"
 	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
 )
 
-func TestNSMGR_RemoteUsecase_Parallel(t *testing.T) {
-	t.Cleanup(func() { goleak.VerifyNone(t) })
+const (
+	nodesCount      = 7
+	suiteTestsCount = 9
+)
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-	defer cancel()
+type nsmgrSuite struct {
+	suite.Suite
 
-	domain := sandbox.NewBuilder(t).
-		SetNodesCount(2).
+	domain *sandbox.Domain
+}
+
+func TestNsmgr(t *testing.T) {
+	suite.Run(t, new(nsmgrSuite))
+}
+
+func (s *nsmgrSuite) SetupSuite() {
+	// Provide 5 seconds per test
```

Comment:
I suppose here we can use simple:
```go
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```
because we have a timeout for each test in suite.